### DISCUSSION TITLE
How to build plugin on MacOS

### DISCUSSION THREAD

**Discussion by linonetwo** [WasmEdge/WasmEdge#2389] [Q&A]
I have done following steps

1. clone the repo
2. install libs `brew install boost cmake ninja llvm@12 openssl`
3. put env into `~/.zshrc` as suggested by brew, for example `export PATH="/usr/local/opt/openssl@3/bin:$PATH"`
4. `cd plugins/wasmedge_httpsreq && mkdir build && cd build && cmake ..` as suggested by ChatGPT
5. get error

```
CMake Error at CMakeLists.txt:7 (wasmedge_add_library):
  Unknown CMake command "wasmedge_add_library".
```

<img width="663" alt="截屏2023-03-30 17 38 25" src="https://user-images.githubusercontent.com/3746270/228795471-0ca20cf9-8e52-4422-9f23-b858457c38c8.png">

**Comment by abhinandanudupa** [comment]
You are going wrong in the step 4. You are moving into the plugin folder but should be moving to the repository root and you are not configuring the build with CMake in the right way.

Follow this guide -  https://wasmedge.org/book/en/contribute/build_from_src/plugin_wasmedge_process.html#build-wasmedge-with-wasmedge-process-plug-in-1 which is for building the Wasmedge Process plugin. 

You set `-DWASMEDGE_PLUGIN_PROCESS=On` to compile the plugins in the plugins folder. If you have your own plugin you might have to write the appropriate CMake files for the same.

**Reply by linonetwo** [reply]
Thanks, I'm trying to install PLUGIN_HTTPSREQ to my Mac dev env.

As you suggested, I do these in WasmEdge repo folder:

1. `mkdir -p build && cd build`
2. `cmake -DCMAKE_BUILD_TYPE=Release -DWASMEDGE_PLUGIN_PROCESS=On -DWASMEDGE_PLUGIN_HTTPSREQ=ON .. && make -j`  (Add `-DWASMEDGE_PLUGIN_HTTPSREQ=ON` because I'm not sure without it I can still get this plugin built)
3. `cmake --install .` as suggested in [plugin_wasmedge_httpsreq.md](https://github.com/WasmEdge/WasmEdge/blob/7fc93df974577941a433faefb7e31141e8ae84cd/docs/book/en/src/contribute/build_from_src/plugin_wasmedge_httpsreq.md) 

But seems httpreq plugin is not installed into my Mac?

```
% cmake --install .
-- Install configuration: "Release"
-- Installing: /usr/local/lib/libwasmedge.0.0.2.dylib
-- Installing: /usr/local/lib/libwasmedge.0.dylib
-- Installing: /usr/local/lib/libwasmedge.dylib
-- Installing: /usr/local/include/wasmedge/wasmedge.h
-- Installing: /usr/local/include/wasmedge/version.h
-- Installing: /usr/local/include/wasmedge/enum.inc
-- Installing: /usr/local/include/wasmedge/enum_configure.h
-- Installing: /usr/local/include/wasmedge/enum_errcode.h
-- Installing: /usr/local/include/wasmedge/enum_types.h
-- Installing: /usr/local/include/wasmedge/int128.h
-- Installing: /usr/local/bin/wasmedgec
-- Installing: /usr/local/bin/wasmedge
```

And I don't know where the artifacts goes

<img width="321" alt="截屏2023-03-31 14 03 00" src="https://user-images.githubusercontent.com/3746270/229036355-f029699d-8704-443a-80bb-0a1bd0c4cba0.png">

The built plug-in path `build/plugins/wasmedge_httpsreq/` is not in the screenshot.
